src/combinatorics/simple_graph/basic.lean

```diff
@@ -122,13 +149,63 @@ set.to_finset G.edge_set
   e ∈ G.edge_finset ↔ e ∈ G.edge_set :=
 by { dunfold edge_finset, simp }
 
+@[simp] lemma edge_set_univ_card [decidable_eq V] [fintype V] [decidable_rel G.adj] :
+  (univ : finset G.edge_set).card = G.edge_finset.card :=
+fintype.card_of_subtype G.edge_finset (mem_edge_finset _)
+
 @[simp] lemma irrefl {v : V} : ¬G.adj v v := G.loopless v
 
 @[symm] lemma edge_symm (u v : V) : G.adj u v ↔ G.adj v u := ⟨λ x, G.sym x, λ x, G.sym x⟩
 
 @[simp] lemma mem_neighbor_set (v w : V) : w ∈ G.neighbor_set v ↔ G.adj v w :=
 by tauto
 
+@[simp] lemma mem_incidence_set (v w : V) : ⟦(v, w)⟧ ∈ G.incidence_set v ↔ G.adj v w :=
+by { simp [incidence_set] }
+
+lemma mem_incidence_iff_neighbor {v w : V} : ⟦(v, w)⟧ ∈ G.incidence_set v ↔ w ∈ G.neighbor_set v :=
+by simp only [mem_incidence_set, mem_neighbor_set]
+
+lemma adj_incidence_set_inter {v : V} {e : sym2 V} (he : e ∈ G.edge_set) (h : v ∈ e) :
+  G.incidence_set v ∩ G.incidence_set h.other = {e} :=
+begin
+  ext e',
+  simp only [incidence_set, set.mem_sep_eq, set.mem_inter_eq, set.mem_singleton_iff],
+  split,
+  { intro h', rw ←sym2.mem_other_spec h,
+    exact (sym2.elems_iff_eq (edge_other_ne G he h).symm).mp ⟨h'.1.2, h'.2.2⟩, },
+  { rintro rfl, use [he, h, he], apply sym2.mem_other_mem, },
+end
+
+section incidence
+variable [decidable_eq V]
+
+/--
+Given an edge incident to a particular vertex, get the other vertex on the edge.
+-/
+def incidence_set_other {v : V} {e : sym2 V} (h : e ∈ G.incidence_set v) : V := h.2.other'
```

Comment:
Should this have `vertex` in the name?

Perhaps
```suggestion
def other_vertex_of_incident {v : V} {e : sym2 V} (h : e ∈ G.incidence_set v) : V := h.2.other'
```
?

Reply:
I think that makes sense